### PR TITLE
ci(): use GitHub actions instead of appveyor & travis

### PR DESCRIPTION
**Pull request by smolck**
ATM only have the publish workflow ported (next is the testing workflow to replace travis), which *should* publish to GitHub packages when run, although I need to test it (on my fork). And assuming we don't care about publishing to NuGet *and* GH packages, then no more dealing with the nuget API key expiring every year, so that's nice.

Some questions on this, I'm assuming we want to run the tests before doing the publishing, right? And when do we want to trigger the workflow? ATM I just have it set to run when triggered manually, but that should be able to be set up to run on pushing a version tag.

### CODE REVIEW

**Comment by justinmk** [comment]
> Some questions on this, I'm assuming we want to run the tests before doing the publishing, right?

yes

> And when do we want to trigger the workflow? ATM I just have it set to run when triggered manually, but that should be able to be set up to run on pushing a version tag.

Manual is fine for now. Let's just be sure to document the "release steps" similar to [here](https://github.com/neovim/pynvim/#release). That makes the surface-area visible, and then we decide whether it makes sense to ever automate these steps. 

Explicit is usually better, at least to start with. Implicit triggers like publish-on-tag can be surprising.

**Comment by smolck** [comment]
Okay, this is ready for review; added the test workflow, the tests are run before the publishing part of the publish workflow, and [got the publish workflow to publish a package finally](https://github.com/smolck/nvim.net/packages/978772) after some trial and error.

Only thing left I guess is documentation on how to publish a package, which is pretty easy, just update the version file in `src/NvimClient.API/NvimClient.API.csproj` and run the workflow.

I can put the above (formatted in steps) at the bottom of the README, sound good? (Note that the last step is obviously only available for maintainers; I guess that could be in the README maybe? not 100% sure . . .)

Also, I'll make a follow-up PR after this swapping out the CI status badges for the new GH actions ones (unless there's a way to do that without merging this first?).

File: src/NvimClient.API/NvimClient.API.csproj
Comment (smolck) [inline src/NvimClient.API/NvimClient.API.csproj:7]: Note I only added these here because this is the only part of the repo that gets published; I could also add these to the others, thoughts?

Comment (justinmk) [reply src/NvimClient.API/NvimClient.API.csproj:7]: could be confusing to have it in places where it's not actually needed/wanted.

**Comment by justinmk** [comment]
> Only thing left I guess is documentation on how to publish a package, which is pretty easy, just update the version file in `src/NvimClient.API/NvimClient.API.csproj` and run the workflow.
> 
> I can put the above (formatted in steps) at the bottom of the README, sound good?

yes

> (Note that the last step is obviously only available for maintainers; I guess that could also be in the README maybe with the steps? not 100% sure . . .)

only maintainers can "release" anyway. Don't need to mention it.

File: .github/workflows/publish.yml
Comment (justinmk) [inline .github/workflows/publish.yml:1]:
```suggestion
name: Publish GitHub NuGet package
```

File: .github/workflows/publish.yml
Comment (justinmk) [inline .github/workflows/publish.yml:4]:
```suggestion
  # no triggers (manual)
  workflow_dispatch:
```

File: README.md
Comment (smolck) [inline README.md:115]: It's a nit but how's this title? `Publishing` could also work, or `Release`, or . . . ?

File: .github/workflows/publish.yml
Comment (justinmk) [inline .github/workflows/publish.yml:36]: any idea why `--store-password-in-clear-text` is needed?

File: README.md
Comment (justinmk) [reply README.md:115]: just `Release`, as in pynvim. We want all release steps to be captured in one place. "Publishing" is over-specific

File: .travis.yml
Comment (justinmk) [inline .travis.yml:12]: the old script ran the generator, do we (plan to) have that in the GHA workflow?

even if it doesn't auto-create a PR or something, it's useful to exercise it I think?

**Review by justinmk — APPROVED** [review]

File: .github/workflows/publish.yml
Comment (smolck) [reply .github/workflows/publish.yml:36]: Not 100% sure, but it's the command they say to use here: https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-nuget-registry#authenticating-with-github_token-in-github-actions

File: .travis.yml
Comment (smolck) [reply .travis.yml:12]: I wondered about this; my thought for leaving it out is this way the file in the published package is the same as the one in the repo, so there aren't any unexpected discrepancies between the two.

Then we can just add another workflow that does update the API and create a PR if we want to automate that.

File: README.md
Comment (smolck) [reply README.md]: Should I rename the workflow from `publish.yml` -> `release.yml` as well, so that's consistent?

**Review by justinmk — APPROVED** [review]

**Comment by smolck** [comment]
Before I merge, thoughts on https://github.com/neovim/nvim.net/pull/14#discussion_r704787352 ?

Also, just a note, once this is merged I'll wait to publish the package until I add the documentation for the public members it's giving warnings about (see [here](https://github.com/neovim/nvim.net/pull/14/files#file-src-nvimclient-api-nvimapi-cs-L22)), so that's available in the release.

File: README.md
Comment (justinmk) [reply README.md]: > Should I rename the workflow from `publish.yml` -> `release.yml` as well, so that's consistent?

up to you. The CI jobs may be necessarily more granular than the README section :)

it's a reversible a decision

**Pull request merged**
